src/cr/cube/cube_slice.py

```diff
@@ -339,63 +334,60 @@ def zscore(self, weighted=True, prune=False, hs_dims=None):
         total = self.margin(weighted=weighted)
         colsum = self.margin(axis=0, weighted=weighted)
         rowsum = self.margin(axis=1, weighted=weighted)
-        res = self._calculate_std_res(
+        zscore = self._calculate_std_res(
             counts, total, colsum, rowsum,
         )
 
         if hs_dims:
-            res = self._intersperse_hs_in_std_res(hs_dims, res)
+            zscore = self._intersperse_hs_in_std_res(hs_dims, zscore)
 
         if prune:
-            arr = self.as_array(
-                prune=prune,
-                include_transforms_for_dims=hs_dims,
-            )
-            if isinstance(arr, np.ma.core.MaskedArray):
-                res = np.ma.masked_array(res, arr.mask)
-
-        return res
+            return self._apply_pruning_mask(zscore, hs_dims)
 
-    def _intersperse_hs_in_std_res(self, hs_dims, res):
-        for dim, inds in enumerate(self.inserted_hs_indices()):
-            for i in inds:
-                if dim not in hs_dims:
-                    continue
-                res = np.insert(res, i, np.nan, axis=(dim - self.ndim))
-        return res
+        return zscore
```

Comment:
Very nice! I can totally see what this method does at a glance, which makes for quick navigation (and also ready unit testing). :heart: